### PR TITLE
Remote update config api design

### PR DESCRIPTION
**Pull request by StarProxima**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new `App Update Checker` package with a comprehensive README outlining the appcast API's structure.
	- Defined a YAML schema for the API that includes customizable links for various app distribution platforms and a detailed releases array for version management, including versioning requirements.

- **Documentation**
	- Added a foundational document to guide developers in implementing the appcast API for checking application updates.
	- Established changelog documentation for tracking project evolution and updates. 

- **Style**
	- Added various `.gitignore` files to maintain a clean repository by excluding unnecessary files for different platforms (iOS, Android, Linux, macOS). 

- **Tests**
	- Created a testing framework for the Runner application, setting the foundation for future unit tests.

### CODE REVIEW

File: app_update_checker/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                "ru": "Добавлены баги, устранены фичи"
+            },
+            // Optional, all by default. Support custom stores
+            "stores": ["googlePlay", "appStore", "ruStore"],
+            // Optional, from stores by default. Support custom platforms
+            "platforms": ["android", "ios", "aurora"]
+        }
+    ]
+}
+```
```
[the remark this answers is not in the document]
Comment (StarProxima) [reply app_update_checker/README.md]: I know, was added for the example. Maybe we should make yaml the default format if it supports comments.

File: app_update_checker/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            },
+            // Optional, will be override
+            "description": {
+                "en": "A new version of $appName is available!\nVersion $version is now available. You have a $currentVersion",
+            },
+            // Optional, may not be displayed
+            "releaseNote": {
+                "en": "Added bugs, fixed features",
+                "es": "Bugs añadidos, correcciones arregladas",
+                "ru": "Добавлены баги, устранены фичи"
```
Comment (crefter) [inline app_update_checker/README.md]: :)

File: app_update_checker/README.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    url: 'https://example.com'
+    platforms:
+      - android
+      - windows
+      - macos
+      - linux
+releases:
+  - version: 0.3.7 # Required
+    isActive: true # Optional, true by default
+    isCritical: false # Optional, false by default
```
Comment (iamgirya) [inline app_update_checker/README.md]: isCritical -> isRequired?

File: app_update_checker/README.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    # Optional, will be override
```
Comment (iamgirya) [inline app_update_checker/README.md:46]: Maybe add isBroken?

File: app_update_checker/README.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    releaseNote: 
+      en: 'Added bugs, fixed features'
+      es: 'Bugs añadidos, correcciones arregladas'
+      ru: 'Добавлены баги, устранены фичи'
+    # Optional, all by default. Support custom stores
+    stores:
+      - googlePlay
+      - appStore
+      - ruStore
+    storesV2:
```
Comment (iamgirya) [inline app_update_checker/README.md]: Like❤️

**Review by iamgirya — CHANGES_REQUESTED** [review]

**Pull request merged**